### PR TITLE
Add metadata functionality to HTTP and S3 drivers

### PR DESCRIPTION
**Pull request by sfc-gh-afedorov**
This allows someone defining an HTTP EF to specify a header `destination-metadata` which picks out data to be returned to the user with the rest of the manifest. Additionally, it allows for pagination to update a value in the `json` parameter sent in the request body.

Tested metadata response on an internal ingestion and pagination parameter via new automated tests in `tests/test_drivers_https_pagination.py` —
1. `test_process_row_pagination_cursor_json`
1. `test_process_row_pagination_cursor_json_null`
1. `test_process_row_pagination_cursor_string`
1. `test_process_row_pagination_cursor_null`
1. `test_process_row_pagination_links`

Tested `destination-metadata` in `tests/tests/test_drivers_https_destination_metadata.py`.

### CODE REVIEW

File: lambda_src/lambda_function.py [hunk 1]
```diff
@@ -194,19 +194,20 @@ def sync_flow(event: Any, context: Any = None) -> Optional[ResponseType]:
         if k.startswith('sf-custom-')
     }
 
-    res_data = process_batch(
+    result = process_batch(
```
Comment (sfc-gh-afedorov) [inline lambda_src/lambda_function.py:197]: the inside of `process_batch` after it calls `process_row` needs to change, as well

File: tests/test_drivers_https_pagination.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        "mock_urlopen",
+        responses,
+    )(test)
+
+
+@mock_urlopen_with_responses(
+    mock_response({'Content-Type': 'application/json'}, b'{"items": [4], "next": "1"}'),
+    mock_response({'Content-Type': 'application/json'}, b'{"items": [2], "next": "2"}'),
+)
+def test_process_row_pagination_cursor_json(mock_urlopen):
```
Comment (sfc-gh-nlele) [inline tests/test_drivers_https_pagination.py:66]: Ooo 🔥🔥

**Comment by sfc-gh-afedorov** [comment]
@sfc-gh-nlele @sfc-gh-pkommini lmk if this looks good or if anything else might be warranted

**Comment by sfc-gh-nlele** [comment]
lgtm!

**Review by sfc-gh-nlele — APPROVED** [review]

**Pull request merged**